internal/policies/mount/handler.go

```diff
@@ -0,0 +1,192 @@
+			logMsg = fmt.Sprintf("Failed to mount %q: %v", m.path, m.err)
+			err = errors.Join(err, fmt.Errorf("failed to mount %q: %w", m.path, m.err))
+		}
+		log.Debugf(ctx, logMsg)
+		i++
+	}
+
+	C.g_main_loop_quit(mainLoop)
+	<-doneMain
+
+	C.g_main_loop_unref(mainLoop)
+	return err
+}
+
+// parseEntries reads the specified file and parses the listed mount locations from it.
+func parseEntries(filepath string) ([]mountEntry, error) {
+	var entries []mountEntry
+
+	content, err := os.ReadFile(filepath)
+	if err != nil {
+		return nil, err
+	}
+
+	lines := strings.Split(string(content), "\n")
+	for _, line := range lines {
+		if line == "" {
+			continue
+		}
+
+		line, krb := strings.CutPrefix(line, "[krb5]")
+		entries = append(entries, mountEntry{path: line, krbAuth: krb})
+	}
+
+	return entries, nil
+}
+
+// setupMountOperation creates and starts a gio mount operation for the specified location.
+// It returns a cleanup function to clean all the allocated C resources.
+func setupMountOperation(entry mountEntry) func() {
+	path := C.CString(entry.path)
+	file := C.g_file_new_for_uri(path)
+
+	op := C.g_mount_operation_new()
+
+	var isAnonymous C.int = C.TRUE
+	if entry.krbAuth {
+		isAnonymous = C.FALSE
+	}
+	C.g_mount_operation_set_anonymous(op, isAnonymous)
+
+	sig := C.CString("ask_password")
+	C.connect_signal(op, sig, C.to_g_callback(C.askPassword), nil)
+
+	C.g_file_mount_enclosing_volume(
+		file,
+		C.GMountMountFlags(0),
+		op,
+		C.g_cancellable_get_current(),
+		C.to_g_async_ready_callback(C.mountDone),
+		nil)
+
+	return func() {
+		C.g_object_unref(C.gpointer(file))
+		C.g_object_unref(C.gpointer(op))
+		C.free(unsafe.Pointer(path))
+		C.free(unsafe.Pointer(sig))
+	}
+}
+
+// askPassword is the callback function that will be called when the mount operation needs a password.
+//
+//export askPassword
+//nolint:revive // We can't use _ for multiple unused parameters in exported functions.
+func askPassword(op *C.GMountOperation, unused1, unused2, unused3 *C.char, flags C.GAskPasswordFlags) {
+	rCode := C.GMountOperationResult(C.G_MOUNT_OPERATION_ABORTED)
+
+	// Checks if the mount operation is anonymous and if the anonymous flag is supported.
+	if C.g_mount_operation_get_anonymous(op) == C.TRUE {
+		if flags&C.G_ASK_PASSWORD_ANONYMOUS_SUPPORTED == C.G_ASK_PASSWORD_ANONYMOUS_SUPPORTED {
+			rCode = C.GMountOperationResult(C.G_MOUNT_OPERATION_HANDLED)
+		}
+	} else {
+		// Otherwise, checks if the kerberos ticket is available.
+		if os.Getenv("KRB5CCNAME") != "" {
+			rCode = C.GMountOperationResult(C.G_MOUNT_OPERATION_HANDLED)
+		}
+	}
+	C.g_mount_operation_reply(op, rCode)
+}
+
+// mountDone is the callback function that will be called when the mount operation is done.
+//
+//export mountDone
+func mountDone(sourceObject *C.GObject, res *C.GAsyncResult, _ C.gpointer) {
+	f := C.to_g_file(sourceObject)
+
+	var err *C.GError
+	C.g_file_mount_enclosing_volume_finish(f, res, &err)
+
+	doneMsg := msg{path: C.GoString(C.g_file_get_uri(f))}
```

Comment:
reading https://docs.gtk.org/gio/method.File.get_uri.html: you need to free the returned value from `g_file_get_uri` or this will be a memory leak.

Reply:
Damn... Tried so hard to avoid those and still have some. I'll fix it.